README.md

```diff
@@ -19,3 +19,109 @@ quantitatively reason with them.
+Take, for instance, if all three (position 0, middle position, end position) are all outside of the desired "good" pivot range. 
+Then the selected pivot from the median would be forced to be out of the good range.
+
+So the visualized permutations of the randomly selected three positions would be: 
+
+3 G - GGG 
+Guaranteed good median, as all three elements selected were within the good range.
+
+2 G - GGB GBG BGG
+Guaranteed good median, as median of two G and one B is always good once sorted, as B must be the first or last sorted value (as it falls either earlier than n/4 or after 3n/4).
+
+1 G - BBG GBB BGB 
+Median is only good if the single G is the middle value after being sorted.
+
+0 G - BBB 
+Guaranteed bad median, as all three elements are bad.
+
+With "G" signifying that a value is within the "good" center range for pivots, and "B" meaning a bad pivot outside of the center range.
+All permutations sharing a row above are in essence the same, as the median pivot selector is assumed to sort the three values taken from the three outlined positions, as they are from a presumably unsorted list and thus have an equal likelihood to be any value.
+As was performed above, these permutations can be grouped by them by how many G/B elements they contain since the values are randomly ordered.
+
+The easiest to consider are the GGG and BBB, corresponding to all three selected indexes holding values that are either all within or outside of the desired range, respectively.
+
+Next, we can consider the permutations with two elements within the n/2 range. 
+This includes GGB GBG and BGG.
+All of these are forced to sort into BGG or GGB, each of which contain a "G" pivot as desired in the median position.
+
+The only permutations that we have not yet considered are BBG, GBB, and BGB.
+These can be sorted into a variety of results. 
+BBG, GBB, and BGB can all sort into either their existing order or the other orders (at equal probability, as each element is of random numerical value).
+
+This can be understood as the "B" elements can be either in the 0 to n/4, or 3n/4 to 1 segments.
+They can both be lower than the good pivot, or they can be both higher than the good pivot in numerical value.
+In each of these scenarios, a bad pivot would ultimately be selected as the median.
+However, the third possibility is that one bad pivot is less than the good pivot, and one is greater than the good pivot.
+This would force the good pivot into the median position, resulting in a good pivot median being selected.
+
+Overall, we can now consider the number of permutations that result in a good median pivot, divided by the overall number of permutations.
+
+There are four cases where the median is guaranteed good. 
+These cases are GGG, GGB, GBG, BGG.
+
+There is one case where the median is guaranteed bad. 
+This case is BBB.
+
+There are three permutations that have a 1/3 probability of having a good pivot selected.
+These cases are GBB, BGB, BBG.
```

Comment:
..but you need to take into account that the good partition is twice as large as either of the bad partitions, i.e. drawing an element from it is twice as likely.